pkg/api/scylla/v1alpha1/types_nodeconfig.go

```diff
@@ -76,6 +76,80 @@ type NodeConfigPlacement struct {
 	NodeSelector map[string]string `json:"nodeSelector"`
 }
 
+// AWSDiscoverer discovers devices on AWS environment.
+// If node setup isn't able to detect whether it's running AWS, it doesn't do the discovery.
+type AWSDiscoverer struct {
+	// disabled controls if discoverer is disabled.
+	Disabled bool `json:"disabled"`
+
+	// deviceRegexp is regular expression used for device discovery.
+	// Discovered block devices must match this regular expression to be passed down to raid array.
+	// +kubebuilder:default:="/dev/nvme\\d+n\\d+$"
+	// +optional
+	DeviceRegexp *string `json:"deviceRegexp"`
+
+	// modelName is a model name of the device. Because multiple types of devices may match the deviceRegexp
+	// it allows to filter out not wanted ones.
+	// +kubebuilder:default:="Amazon EC2 NVMe Instance Storage"
+	// +optional
+	ModelName *string `json:"modelName"`
+}
+
+// StaticDiscoverer discovers devices from predefined list of devices.
+type StaticDiscoverer struct {
+	// disabled controls if discoverer is disabled.
+	Disabled bool `json:"disabled"`
+
+	// staticDevices is a list of devices injected into discovered devices.
+	// These devices must exist on every node matched by this NodeConfig.
+	// +optional
+	StaticDevices []string `json:"staticDevices,omitempty"`
+}
+
+type Discoverer struct {
+	// aws contains configuration options for AWS discoverer.
+	// +optional
+	AWS *AWSDiscoverer `json:"aws"`
+
+	// aws contains configuration options for AWS discoverer.
+	// +optional
+	Static *StaticDiscoverer `json:"static"`
+}
+
+type Filesystem string
+
+const (
+	XFSFilesystem Filesystem = "xfs"
+)
+
+type MountOptions struct {
+	// hostPath is a path on the host where formatted array will be mounted.
+	// +kubebuilder:validation:Required
+	HostPath string `json:"hostPath"`
+
+	// options is a list of options used during array mounting.
+	// +optional
+	Options []string `json:"options,omitempty"`
+}
+
+type LocalDiskSetup struct {
+	// disabled controls if automatic local disk setup is disabled.
+	// +optional
+	Disabled bool `json:"disabled"`
+
+	// discoverer defines parameters of different discoverers searching for locally attached disks.
+	// All discovered devices are combined into single RAID0 array, formatted to provided Filesystem and
+	// mounted at using Mount options.
+	Discoverer Discoverer `json:"discoverer"`
```

Comment:
Maybe we could simplify / reorient the view to how user sees it, what do you think about this?
```yaml
compositions: # needs a better name
- type: RAID0
  RAID0:
  # in theory we could have this selector per raid group but that can be extended on but mostly people don't care which one is in group A and which is in group B
  devices: 
    nameRegex: "/dev/nvme*"
    modelRegex: "Micron 2300 NVMe .*"
  name: my-raid-nvme
mounts:
- device: /dev/md/my-raid-nvme
  mountPoint: /mnt/my-nvme
fileSystems:
- device: /dev/md/my-raid-nvme
  type: xfs
  xfsOptions:
  - foo
```

Reply:
I like it better, actually, it will allow us to set up more complex setups like multiple raids on with different configurations. For example RAID0 out of local nvme, then raid1 with NAS for persistance.
I renamed some of the fields, what do you think?
```
raids:
- name: my-raid-nvme
  type: RAID0
  RAID0:
    devices:
      nameRegex: "/dev/nvme*"
      modelRegex: "Micron 2300 NVMe .*"
fileSystems:
- raidName: my-raid-nvme
  type: xfs
mounts:
- raidName: my-raid-nvme
  mountHostPath: /mnt/my-nvme

Reply:
i don't think the other 2 should be connected to raid, you can create fs on other things like LVM or cloud block devices that are already redundant

Reply:
note: I've also used naming for the arguments close or the same as the Linux tools use so there is no ambiguity

Reply:
```
raids:
- name: my-raid-nvme
  type: RAID0
  RAID0:
    devices: 
      nameRegex: "/dev/nvme*"
      modelRegex: "Micron 2300 NVMe .*"
fileSystems:
- device: /dev/md/my-raid-nvme
  type: xfs
mounts:
- device: /dev/md/my-raid-nvme
  mountPoint: /mnt/my-nvme
  unsupportedOptions:
  - prjquota

Reply:
api looks good, I'd defer the mount options naming struct / based on where the mount persistent goes as it may no longer be the command line args for mount